x-pack/metricbeat/module/prometheus/collector/histogram.go

```diff
@@ -0,0 +1,52 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package collector
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+// promHistogramToES takes a Prometheus histogram and converts it to an ES histogram
+func promHistogramToES(cc CounterCache, name string, labels common.MapStr, histogram *dto.Histogram) common.MapStr {
+	var values []float64
+	var counts []uint64
+
+	// calculate centroids and rated counts
+	var lastUpper, prevUpper float64
+	var sumCount uint64
+	for _, bucket := range histogram.GetBucket() {
+		// Ignore non-numbers
+		if bucket.GetCumulativeCount() == uint64(math.NaN()) || bucket.GetCumulativeCount() == uint64(math.Inf(0)) {
+			continue
+		}
+
+		prevUpper = lastUpper
+		if bucket.GetUpperBound() == math.Inf(0) {
+			// Report +Inf bucket as a point, interpolating its value
+			values = append(values, lastUpper+(lastUpper-prevUpper))
```

Comment:
`lastUpper-prevUpper == 0` here? What is the reason of adding this explicitly?

Reply:
ups you are right, last bucket math is wrong, I'm fixing it

Reply:
👍